judge-d-server/src/main/java/com/hltech/judged/server/domain/contracts/ServiceContracts.java

```diff
@@ -14,21 +15,28 @@
 import static java.util.stream.Collectors.toMap;
 
 @Getter
-@AllArgsConstructor
 @ToString
-@EqualsAndHashCode
+@EqualsAndHashCode(exclude = "publicationTime")
```

Comment:
IMO We should not include publication time in hash code - if we will try to add the same contracts for the same service couple times it will be still the same contract, time just will be updated, publication time is more like metadata than business value

Reply:
It's a business metadata as from the Judged business' point of view it is required :). But of course you are right that it should not be taken into the account when comparing the contracts i.e. it is not a part of identity.
In general it would be extra nice to have a separate representation of metadata added to each contract. We have a task to add other data like tags. We can prepare the code for that in another task. (can you please verify we have a task for metadata storing and handling)?

Reply:
Issue added https://github.com/HLTech/judge-d/issues/180